WorkloadManagementSystem/Utilities/RabbitMQSynchronizer.py

```diff
@@ -0,0 +1,136 @@
+'''RabbitMQSynchronizer
+
+  RabbitMQSynchronizer keeps the RabbitMQ user database  synchronized with the CS
+  RabbitMQ user database is updated according to changes in CS.
+'''
+
+from DIRAC import S_OK
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import hostHasProperties
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getDNsInGroup
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getDNForHost
+from DIRAC.ConfigurationSystem.Client.Helpers.Registry import getHosts
+from DIRAC.Core.Utilities.RabbitMQAdmin import getAllUsers, deleteUsers
+from DIRAC.Core.Utilities.RabbitMQAdmin import setUsersPermissions, addUsersWithoutPasswords
+from DIRAC.ResourceStatusSystem.Utilities import CSHelpers
+from DIRAC import gLogger
+from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations #for getValidPilotGroupName()
+
+class RabbitMQSynchronizer(object):
+
+  def __init__( self ):
+
+    # Warm up local CS
+    # I am not sure whether it is needed but
+    # it was used in DIRAC.ResourceStatusSystem.Utilities.Synchronizer
+    CSHelpers.warmUp()
+    self._accessUserGroup = getAllowedGroupName() #only users belonging to group with this property are allowed to connect
+    self._accessProperty = getAllowedHostProperty() #only host with this property are allowed to connect
+
+  def sync( self, _eventName, _params ):
+    '''Synchronizes the internal RabbitMQ user database with the current content of CS.
+
+    Args:
+      _eventName: any value, this parameter is ignored, but needed by caller function.
+      _params: - any value, this parameter is ignored, but needed by caller function.
+    Returns:
+      S_OK:
+    Example:
+       s.sync( None, None )
+    '''
+
+    valid_users = getDNsInGroup(self._accessUserGroup)
+    valid_hosts = getDNsForValidHosts(self._accessProperty)
+    updateRabbitMQDatabase(valid_users + valid_hosts)
+    return S_OK()
+
+def getAllowedGroupName():
```

Comment:
for all these public functions, the return type should eb DIRAC compliant. If you only need them here, make them private static method